manifests/database.pp

```diff
@@ -7,26 +7,31 @@
       $mongodb_pidfilepath = '/var/run/mongodb/mongodb.pid'
     }
 
-    $auth_real = $::pulp::db_username != undef and $::pulp::db_password != undef
+    # puppetlabs-mongodb is totally broken with managing users on < 2.6
+    if (versioncmp($::mongodb_version, '2.6.0') >= 0) {
+      $auth_real = $::pulp::db_username != undef and $::pulp::db_password != undef
+
+      if $auth_real {
+        mongodb_user { $::pulp::db_username:
+          ensure        => present,
+          username      => $::pulp::db_username,
+          password_hash => mongodb_password($::pulp::db_username, $::pulp::db_password),
+          database      => $::pulp::db_name,
+          roles         => ['dbAdmin', 'readWrite'],
+          tries         => 10,
+          require       => Service['mongodb'],
+        }
+      }
+    } else {
+      $auth_real = false
+    }
 
     class { '::mongodb::server':
       pidfilepath => $mongodb_pidfilepath,
       auth        => $auth_real,
       noauth      => !$auth_real,
     }
 
-    if $auth_real {
```

Comment:
It looked better to me to have all the logic in one block, so you know why we're doing this, but I can move it back if you'd like